src/views/self/view_teleport.cpp

```diff
@@ -93,52 +93,76 @@ namespace big
 
 		ImGui::SeparatorText("GUI_TAB_IPL"_T.data());
 
-		if (ImGui::BeginCombo("IPL_LOCATION"_T.data(), ipls[g.self.ipls.select].friendly_name))
+		static int current_select = g.self.ipls.select;
+		static int last_select    = current_select;
+
+		ImGui::SetNextItemWidth(400);
+		if (ImGui::BeginCombo("##Ipllocation", ipls[current_select].friendly_name))
 		{
 			for (int i = 0; i < IM_ARRAYSIZE(ipls); i++)
 			{
-				if (ImGui::Selectable(ipls[i].friendly_name, i == g.self.ipls.select))
-					g.self.ipls.select = i;
-
-				if (i == g.self.ipls.select)
+				bool is_selected = (i == current_select);
+				if (ImGui::Selectable(ipls[i].friendly_name, is_selected))
+				{
+					current_select = i;
+				}
+				if (is_selected)
+				{
 					ImGui::SetItemDefaultFocus();
+				}
 			}
-
 			ImGui::EndCombo();
 		}
-
-		const auto& selected_ipl = ipls[g.self.ipls.select];
-		if (components::button("LOAD_IPL"_T))
+		ImGui::SameLine();
+		components::button("LOAD_IPL"_T, [] 
 		{
-			//unload all previous ipls
-			for (auto& ipl : ipls)
-				for (auto& ipl_name : ipl.ipl_names)
+			// If we've changed selections, first unload previously loaded IPL, then load previously deleted IPLs
+			if (current_select != last_select)
+			{
+				// Unload previously loaded IPL of the last selection
+				for (auto& ipl_name_unload : ipls[last_select].ipl_names)
 				{
-					if (STREAMING::IS_IPL_ACTIVE(ipl_name))
+					if (STREAMING::IS_IPL_ACTIVE(ipl_name_unload))
 					{
-						LOG(INFO) << "unloading existing ipl " << ipl_name;
-						STREAMING::REMOVE_IPL(ipl_name);
+						STREAMING::REMOVE_IPL(ipl_name_unload);
+						LOG(INFO) << "Unloaded previous IPL " << ipl_name_unload;
 					}
 				}
 
-			//load the new ipl
-			for (auto& ipl_name : selected_ipl.ipl_names)
-				STREAMING::REQUEST_IPL(ipl_name);
-		}
+				// Load previously deleted IPLs of the last selection
+				for (auto& ipl_name_load : ipls[last_select].ipl_names_remove)
+				{
+					STREAMING::REQUEST_IPL(ipl_name_load);
+					LOG(INFO) << "Loaded previously deleted IPLs " << ipl_name_load;
+				}
 
-		ImGui::SameLine();
+				// Load new IPLs of the current selection
+				for (auto& ipl_name : ipls[current_select].ipl_names)
+				{
+					STREAMING::REQUEST_IPL(ipl_name);
+					LOG(INFO) << "New IPL loaded " << ipl_name;
+				}
+
+				// Remove old IPLs of the current selection to avoid conflicts
+				for (auto& ipl_name_remove : ipls[current_select].ipl_names_remove_when_load)
+				{
+					STREAMING::REMOVE_IPL(ipl_name_remove);
+					LOG(INFO) << "Removed IPL to avoid conflict " << ipl_name_remove;
```

Comment:
These logs are likely to cause spam, best to remove these before adding it to master.

Reply:
solved